scene/main/http_request.cpp

```diff
@@ -240,7 +240,7 @@ bool HTTPRequest::_handle_response(bool *ret_value) {
 		String new_request;
 
 		for (const String &E : rheaders) {
-			if (E.containsn("Location: ")) {
+			if (E.to_lower().begins_with("location: ")) {
 				new_request = E.substr(9).strip_edges();
 			}
```

Comment:
```suggestion
			if (E.to_lower().begins_with("location: ")) {
				new_request = E.substr(9).strip_edges();
				break;
			}
```
We can stop parsing headers if we found the first `Location` header, as the HTTP spec seems to disallow multiple such headers in the same response.

Also, further down in line 253, we check `if (new_request.begins_with("http"))` to determine whether the URL is relative or absolute to determine whether this should be a new request? I haven't looked at this more than reviewing this PR and its quite late here, so I might be missing things, but if true, there are two suspect things here for me:

1. `http` alone is strictly speaking not enough to determine if the following URL is absolute. We may want to check whether the request begins with `http://` or `https://`.
2. At first glance, this check seems to be case sensitive, which is against [spec](https://www.rfc-editor.org/rfc/rfc3986#section-3.1): `An implementation
   should accept uppercase letters as equivalent to lowercase in scheme
   names (e.g., allow "HTTP" as well as "http") for the sake of
   robustness but should only produce lowercase scheme names for
   consistency.`

Reply:
Given this is already how the code was structured, this can be handled in a separate PR